test/e2e/work_load_test.go

```diff
@@ -320,5 +323,129 @@ var _ = Describe("workload orchestration testing", func() {
+						Reason: "ApplySucceeded",
+						Status: metav1.ConditionTrue,
+						Type:   string(v1alpha1.ResourcePlacementStatusConditionTypeApplied),
+					},
+				},
+				SelectedResources: []v1alpha1.ResourceIdentifier{
+					{
+						Version:   "v1",
+						Kind:      "Secret",
+						Name:      testSmallSecret.Name,
+						Namespace: testSmallSecret.Namespace,
+					},
+					{
+						Version: "v1",
+						Kind:    "Namespace",
+						Name:    namespace.Name,
+					},
+				},
+				TargetClusters: []string{"kind-member-testing"},
+			}
+			utils.WaitCreateClusterResourcePlacementStatus(ctx, *HubCluster, &types.NamespacedName{Name: crp.Name}, crpStatus, crpStatusCmpOptions, 3*utils.PollTimeout)
+
+			By("check if resources in namespace are propagated to member cluster")
+			ownerReferences := []metav1.OwnerReference{
+				{
+					APIVersion:         workapi.GroupVersion.String(),
+					BlockOwnerDeletion: pointer.Bool(false),
+					Kind:               "AppliedWork",
+					Name:               crp.Name,
+				},
+			}
+			expectedNamespace := namespace
+			expectedSecret := &testSmallSecret
+			expectedNamespace.OwnerReferences = ownerReferences
+			expectedSecret.OwnerReferences = ownerReferences
+
+			utils.CmpNamespace(ctx, *MemberCluster, &types.NamespacedName{Name: namespace.Name}, expectedNamespace, resourceIgnoreOptions)
+			// Ignoring Annotations here because fleet.azure.com/last-applied-configuration has live fields, checking to see if it's not empty instead.
+			gotSecret := utils.CmpSecret(ctx, *MemberCluster, &types.NamespacedName{Name: testSmallSecret.Name, Namespace: testSmallSecret.Namespace}, expectedSecret, resourceIgnoreOptions)
+			Expect(gotSecret.Annotations[workcontroller.LastAppliedConfigAnnotation]).To(Not(BeEmpty()))
+			testSmallSecretSpecHash := gotSecret.Annotations[workcontroller.ManifestHashAnnotation]
+
+			By("update secret so that annotation limit crosses threshold of 256KB")
+			var testLargeSecret corev1.Secret
+			err = pkgutils.GetObjectFromManifest("./test/integration/manifests/resources/test-large-secret.yaml", &testLargeSecret)
+			Expect(err).Should(Succeed())
+			// testLargeSecret has the same name as testSmallSecret
+			Expect(HubCluster.KubeClient.Update(ctx, &testLargeSecret)).Should(Succeed(), "Failed to update secret %s to be large in %s cluster", testLargeSecret.Name, HubCluster.ClusterName)
+			expectedSecret = &testLargeSecret
+			expectedSecret.OwnerReferences = ownerReferences
+
+			// Ignoring Annotations here because fleet.azure.com/last-applied-configuration has live fields, checking to see if it's not empty instead.
+			gotSecret = utils.CmpSecret(ctx, *MemberCluster, &types.NamespacedName{Name: testLargeSecret.Name, Namespace: testLargeSecret.Namespace}, expectedSecret, resourceIgnoreOptions)
+			Expect(gotSecret.Annotations[workcontroller.LastAppliedConfigAnnotation]).To(BeEmpty())
+			Expect(gotSecret.Annotations[workcontroller.ManifestHashAnnotation]).ToNot(Equal(testSmallSecretSpecHash))
```

Comment:
did not see much value to compare with the smallSecretSpecHash. If you want to validate this annotation, you need to compare the exact value.